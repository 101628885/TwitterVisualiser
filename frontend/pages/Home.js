/**
|--------------------------------------------------
| Home page
|--------------------------------------------------
*/

import React from 'react';
import {
    FlatList,
    ActivityIndicator,
    Text,
    View,
    StyleSheet
} from 'react-native';
import {
    Container,
    H2,
    Header,
    Title,
    Content,
    Button,
    Icon,
    Body,
    Left,
    Toast,
    Root,
    Right
} from "native-base";
import Font from 'expo';

export default class Home extends React.Component {
    constructor(props) {
        super(props);
        this.state = { dataLoaded: false, fontLoaded: false }
    }

    static navigationOptions = {
        drawerIcon: (
            <Icon name="home" style={{ color: "#0084b4" }} />
        )
    }

    // Part of the react lifecyle
    async componentDidMount() {
        // Use if server down
        // await fetch('https://testtwitter-diigzuppaq.now.sh/shanesAndCoreysSpecialEndPoint/10')
        await fetch('http://144.6.226.34:3000/shanesAndCoreysSpecialEndPoint/10')
            .then((res) => res.json())
            .then((resJson) => {
                this.setState({
                    dataLoaded: true,
                    dataSource: resJson,
                });
            })
            .catch((error) => {
                console.error(error);
            });
    }

    // Part of the react lifecyle
    async componentWillMount() {
        await Expo.Font.loadAsync({
            'Roboto': require('native-base/Fonts/Roboto.ttf'),
            'Roboto_medium': require('native-base/Fonts/Roboto_medium.ttf'),
        });
        this.setState({ fontLoaded: true });
    }

    render() {
        return (
            this.state.dataLoaded && this.state.fontLoaded ?
                <Container style={style.container}>
                    <Header>
                        <Left>
                            <Button
                                transparent
                                onPress={() => this.props.navigation.navigate("DrawerOpen")}>
                                <Icon name="menu" />
                            </Button>
                        </Left>
                        <Body>
                            <Title style={{ color: "#0084b4", alignSelf: 'center' }}>Home</Title>
                        </Body>
                        <Right />
                    </Header>
                        <View style = { style.titleView }>
                            <H2 style = { style.title }>Welcome to Twiiter Visualiser</H2>
                        </View>
                </Container>
                :
                <View style={style.mb}>
                    <ActivityIndicator />
                </View>
        )
    }
}

const style = StyleSheet.create({
    container: {
        backgroundColor: "#0084b4",
    },
    mb: {
        marginBottom: 15,
        padding: 20,
    },
    title: {
        fontStyle: 'italic',
        color: '#FFF',
    },
    titleView: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    }
})