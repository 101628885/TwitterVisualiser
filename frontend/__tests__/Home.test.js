import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Home from '../pages/Home';

describe('Home', () => {
  it('renders correctly', () => {
    const output = shallow(
      <Home />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
