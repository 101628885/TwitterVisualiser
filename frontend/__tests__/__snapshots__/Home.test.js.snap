// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
      "padding": 20,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
